Rename deck

EthanUphoff opened the pull request:
closes #20 
Allows deck owners and teachers to rename decks.

| ![editbutton](https://user-images.githubusercontent.com/25913052/33521111-3bfb0b50-d78e-11e7-9933-fa9c55e92e46.png) | ![renamingeditor](https://user-images.githubusercontent.com/25913052/33521113-3f1d4fb4-d78e-11e7-8c59-92fd3ebdf1c1.png) | 
|---|---|

![finalrename](https://user-images.githubusercontent.com/25913052/33521114-41fa1460-d78e-11e7-9b29-7dcb8f5774fd.png)

EthanUphoff committed:
Adds Deck Name Editing

Adds deck name editing for deck owners and those with the ability to edit.

EthanUphoff committed:
Makes it so only deck owners and teachers can change deck name

This makes it so students can no longer change the name of decks even if they have the ability to edit.

EthanUphoff committed:
Did some minor UI changes

Makes the rename button a raised button that is disabled when there is no text in the input and the edit button is more centered now.

floogulinc commented:
The spacing looks all messed up now, especially on the desktop screenshot

EthanUphoff committed:
Slight change for consistency

Made this to match the way that class renaming looks.

floogulinc reviewed (changes requested):
The menu icon shouldn't move when editing a card on mobile as it does in the screenshot.

EthanUphoff committed:
Slight change for consistency

Made this to match the way the snackbar displays to match the way the class renaming snackbar is displayed

floogulinc committed:
Correct snackbar text

EthanUphoff committed:
Further changes for consistency

Changes the save button to an icon button.

floogulinc committed:
Merge branch 'master' into rename-deck

floogulinc committed:
Update rename deck to use new EditableText component

floogulinc commented:
Long titles still mess up the deck name:
![](https://vgy.me/UcDxB1.png)
